src/EntryPoints/Specials/SpecialPendingApprovals.php

```diff
@@ -22,11 +24,21 @@ public function execute( $subPage ): void {
 		$this->checkPermissions();
 		$this->checkReadOnly();
 
-		$this->getOutput()->addHTML(
-			$this->createPendingApprovalsTable(
-				$this->pendingApprovalRetriever->getPendingApprovalsForApprover( $this->getUser()->getId() )
-			)
-		);
+		$categories = $this->approverRepository->getApproverCategories( $this->getUser()->getId() );
+
+		if ( $categories === [] ) {
+			$this->getOutput()->addWikiMsg( 'pageapprovals-no-categories' );
```

Comment:
The MW exception `PermissionsError` won't work here, because it's not strictly a right-related permission error.

Reply:
We could maybe use the `onSpecialPageInitList` hook to unregister the page when the user has no categories, but that can be investigated later.